src/api/system.ts

```diff
@@ -34,6 +37,50 @@ export class System {
         });
     }
 
+    waitFor<Type extends keyof WaitForResponse>(
+        jobType: Type,
+        id: string,
+        options: { interval?: number; maxRetries?: number } = {}
+    ): D2ApiResponse<WaitForResponse[Type] | null> {
+        const { interval = 1000, maxRetries } = options;
+
+        let isCancel = false;
+        let retries = 0;
+
+        const checkTask = async () => {
+            const response = await this.d2Api
+                .get<{ message: string; completed?: boolean }[]>(`/system/tasks/${jobType}/${id}`)
+                .getData();
+
+            const result = response || [];
```

Comment:
[minor] according to the type, we don't need `|| []`, it may be undefined or null? we should add that possibility also to the type. In any case the check `!result! in the next line is always true, right?

Reply:
```ts
            const checkTask = async () => {
            const result = await this.d2Api
                .get<{ message: string; completed?: boolean }[]>(`/system/tasks/${jobType}/${id}`)
                .getData();

            if (!Array.isArray(result) || !result[0]) return false;
            return result[0].completed;
        };
```